app/services/publish_service.rb

```diff
@@ -29,6 +31,20 @@ def publish_assets(live_edition)
     PublishAssetService.call(edition, live_edition)
   end
 
+  def associate_with_government
+    return if edition.government
+
+    date = edition.backdated_to || edition.document.first_published_at || Time.current
+    government = Government.for_date(date)
+    edition.assign_attributes(government_id: government&.content_id)
```

Comment:
It feels a bit odd to have almost identical code in two places. I understand why it's being done, but the use of `Time.current` to find the appropriate government is a very subtle difference, and I worry about it being missed by future developers.

Could there be case for creating a `AssociateWithGovernmentService` that allows documents flagged as new at the time of calling to use `Time.current`?

Reply:
Sure, I understand your concerns. I wanted to avoid a situation where we have overly specific methods for this e.g. `Edition#date_for_government` or a service that has to deal with the `Time.current` edge case (which is how we got `fallback_government` last time around.

In the end since we only have this logic twice and it's slightly different both times and ultimately quite simple I felt it was ok to duplicate it. I'm happy to address this if we hit a situation where we have it another time.